src/istioctl/istioctl.go

```diff
@@ -203,84 +201,10 @@ func ExecWithWriters(homeDir string, args []string, stdout, stderr io.Writer) er
 	return cmd.Run()
 }
 
-func Fetch(homeDir string, version, flavor string, flavorVersion int, ms *api.Manifest) (*api.IstioDistribution, error) {
-	d, err := processFetchParams(version, flavor, flavorVersion, ms)
-	if err != nil {
-		return nil, err
-	}
-
+func Fetch(homeDir string, d *api.IstioDistribution, ms *api.Manifest) (*api.IstioDistribution, error) {
```

Comment:
I think we no longer need to return distro

```suggestion
func Fetch(homeDir string, target *api.IstioDistribution, ms *api.Manifest) error {
```